dll/win32/shell32/CShellItem.cpp

```diff
@@ -335,3 +335,98 @@ HRESULT WINAPI SHCreateShellItem(PCIDLIST_ABSOLUTE pidlParent,
 
     return hr;
 }
+
+class CShellItemArray :
+    public CComCoClass<CShellItemArray, &CLSID_NULL>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IShellItemArray
+{
+    CIDA *m_pCIDA;
+    STGMEDIUM m_Medium;
+
+public:
+    CShellItemArray() : m_pCIDA(NULL)
+    {
+        m_Medium.tymed = TYMED_NULL;
+    }
+
+    virtual ~CShellItemArray()
+    {
+        CDataObjectHIDA::DestroyCIDA(m_pCIDA, m_Medium);
+    }
+
+    HRESULT Initialize(IDataObject *pdo)
+    {
+        return CDataObjectHIDA::CreateCIDA(pdo, &m_pCIDA, m_Medium);
+    }
+
+    inline UINT GetCount() const { return m_pCIDA->cidl; }
+
+    // IShellItemArray
```

Comment:
Add `override` keyword for all the overriding methods.

Reply:
I have been told not to do that.

Reply:
Then, add that keywords and build to test integrity, and them erase them.

@JoachimHenze I blame you for integrity check difficulty.

Reply:
Don't listen to @JoachimHenze, this is C++ code not C code. You should always use the override keyword when you override a function. Inheritance is fundamental to how Windows operates.

Reply:
yeah I'm pretty sure that's C++ behavior, not sure why anyone would say otherwise

Reply:
> yeah I'm pretty sure that's C++ behavior

Yes but the `override` keyword is C++ whatever version and I suppose it makes it harder to port it back to whatever older compiler does not have it.

Reply:
We don't support older compilers and we don't support backporting. It is just a hobby for @JoachimHenze.

Reply:
Allow me one last question at least @learn-more :
In which header would it be most reasonable to place a
```
if defined(_MSC_VER)
#if (_MSC_VER <= 1600) // Visual Studio 2010
#define override
#endif
#endif
```
in the older branches to give maximum coverage over the ReactOS project

Reply:
@JoachimHenze probably something nasty / unrelated like `include/psdk/basetyps.h`.
I am unsure why I am the only one that is able to answer this though?

Reply:
Thank you. I asked **you**, because I expected the answer from you to be a well-educated one. Resolved.